MTM-53068 improvements and refresh in microservice sdk docs. (#1017)
SoftwareAG/c8y-docs: content/microservice-sdk/concept-bundle/security.md

1. The microservice can be created in any tenant that has **feature-microservice-hosting** enabled.
2. The microservices access the tenant API.

> At the installation time of the microservice, an application is created in the owner tenant (usually "super-tenant") reflecting the new microservice. In addition, a bootstrap user is created in the owner tenant that allows the microservice to retrieve subscriptions.

Reviewer: We do not use the term "super tenant" as this is unclear. Could we say "usually the {{< management-tenant >}}"?

Author: well, somebody told me "super-tenant" is "management or EE tenant" when I wanted to remove it from another place in the code. The problem with previous explanation was a claim that only management can create microservices, and in fact every tenant with subscibed feature-microservice-hosting can do it. Many EE tenants use this feature and microservices are subscribed to their subtenants.

Maybe 
`(usually management or Enterprise Edition tenant)` 
?

Reviewer: I would then phrase it "usually a {{< management-tenant >}} or {{< enterprise-tenant >}}". We use short codes for the tenant types. Btw, it is just Enterprise tenant, not Enterprise edition tenant :-).

I will change the terminology in the other place as well then, thanks for the link. We should be really clear with these terms.

Whenever required, a platform administrator will subscribe a customer to the new microservice. As part of the subscription, a service user in the customer tenant is created using random credentials.

#### Microservice authorization

Authorization is relevant on two levels:

1. On the owner tenant level, the only authorization of a microservice is to access its own subscriptions.
2. For accessing customer tenants, the microservice installs a set of required permissions for being able to operate.

A microservice is associated with a bootstrap user in the owner tenant, which will make sure that only its subscriptions are returned. A microservice is also associated with a set of permissions that it requires for carrying out its function on a customer tenant.

> These permissions are visualized in the Administration application. The permissions are associated with the service user that is created when a platform administration associates a microservice with a tenant (subscribes it). Owner tenant also needs to be subscribed if you intend to use microservice features on this tenant. 

Reviewer 2: ```suggestion
These permissions are visualized in the Administration application. The permissions are associated with the service user that is created when a platform administration associates a microservice with a tenant (subscribes to it). The owner tenant must also be subscribed to if you intend to use microservice features on it. 
```
